internal/controller/serverclaim_controller_test.go

```diff
@@ -244,13 +243,19 @@ var _ = Describe("ServerClaim Controller", func() {
 		Expect(k8sClient.Create(ctx, claim)).To(Succeed())
 
 		By("Patching the Server to available state")
-		Eventually(UpdateStatus(&server, func() {
+		Eventually(UpdateStatus(server, func() {
 			server.Status.State = metalv1alpha1.ServerStateAvailable
 		})).Should(Succeed())
 
 		By("Ensuring that the Server has the correct claim ref")
-		Eventually(Object(&server)).Should(SatisfyAll(
-			HaveField("Spec.ServerClaimRef.Name", claim.Name),
+		Eventually(Object(server)).Should(SatisfyAll(
```

Comment:
@defo89 that was the reason for the sporadic test failures which we encountered. Eventually `.spec.ServerClaimRef.Name` was set but if not the test ran into a `nil` pointer deref.